src/devices/Arduino/ArduinoBoard.cs

```diff
@@ -0,0 +1,177 @@
+
+namespace Iot.Device.Arduino
+{
+    /// <summary>
+    /// Implements an interface to an arduino board which is running Firmata.
+    /// See documentation on how to prepare your arduino board to work with this.
+    /// Note that the program will run on the PC, so you cannot disconnect the
+    /// Arduino while this driver is connected.
+    /// </summary>
+    public class ArduinoBoard : IDisposable
+    {
+        private Stream _serialPortStream;
+        private FirmataDevice _firmata;
+        private Version _firmwareVersion;
+        private string _firmwareName;
+        private List<SupportedPinConfiguration> _supportedPinConfigurations;
+
+        public ArduinoBoard(Stream serialPortStream)
+        {
+            _serialPortStream = serialPortStream;
+        }
+
+        public event Action<string, Exception> LogMessages;
+
+        public virtual void Initialize()
+        {
+            _firmata = new FirmataDevice();
+            _firmata.Open(_serialPortStream);
+            _firmata.OnError += FirmataOnError;
+            var protocolVersion = _firmata.QueryFirmataVersion();
+            if (protocolVersion != _firmata.QuerySupportedFirmataVersion())
+            {
+                throw new NotSupportedException($"Firmata version on board is {protocolVersion}. Expected {_firmata.QuerySupportedFirmataVersion()}. They must be equal.");
+            }
+
+            Log($"Firmata version on board is {protocolVersion}.");
+
+            _firmwareVersion = _firmata.QueryFirmwareVersion(out _firmwareName);
+
+            Log($"Firmware version on board is {_firmwareVersion}");
+
+            _firmata.QueryCapabilities();
+
+            _supportedPinConfigurations = _firmata.PinConfigurations; // Clone reference
+
+            Log("Device capabilities: ");
+            foreach (var pin in _supportedPinConfigurations)
+            {
+                Log(pin.ToString());
+            }
+
+            // _firmata.SetSamplingInterval(TimeSpan.FromMilliseconds(100));
+            _firmata.EnableDigitalReporting();
+        }
+
+        public Version FirmwareVersion
+        {
+            get
+            {
+                return _firmwareVersion;
+            }
+        }
+
+        public string FirmwareName
+        {
+            get
+            {
+                return _firmwareName;
+            }
+        }
+
+        internal FirmataDevice Firmata
+        {
+            get
+            {
+                return _firmata;
+            }
+        }
+
+        internal List<SupportedPinConfiguration> SupportedPinConfigurations
+        {
+            get
+            {
+                return _supportedPinConfigurations;
+            }
+        }
+
+        internal void Log(string message)
+        {
+            LogMessages?.Invoke(message, null);
+        }
+
+        private void FirmataOnError(string message, Exception innerException)
+        {
+            LogMessages?.Invoke(message, innerException);
```

Comment:
This should probably call Log()

Reply:
For log, I've added a logger solution because I needed it for the NFC reader development. See https://github.com/dotnet/iot/blob/master/src/devices/Card/LogInfo.cs, I guess you can reuse the logger.

Reply:
Sounds good, but I guess we should move it to the core then.

Reply:
Agree. I let @joperezr comments on this

Reply:
I think sorting out the logging stuff should be on a separate PR, so I'll leave as-is for now.